### PR TITLE
feat: improve hardware wallet setup on onboarding flow

### PR DESCRIPTION
**Pull request by greatertomi**
# Checklist

- [x] JIRA - LW-9809
- [ ] Proper tests implemented
- [x] Screenshots added.

---

## Proposed solution

Update the account selection screen and remove the "name your wallet" step on the hardware wallet onboarding flow 

## Testing

Describe here, how the new implementation can be tested.
Provide link or briefly describe User Acceptance Criteria/Tests that need to be met

## Screenshots

Screenshot 2024-02-22 at 14 07 14

### CODE REVIEW

File: apps/browser-extension-wallet/src/views/browser-view/features/wallet-setup/components/HardwareWalletFlow.tsx [hunk 1]
```diff
@@ -8,8 +8,7 @@ import {
   WalletSetupLegalStep,
   WalletSetupFinalStep,
   WalletSetupConnectHardwareWalletStep,
-  WalletSetupSelectAccountsStep,
-  WalletSetupWalletNameStep
+  RevampedWalletSetupSelectAccountsStep
```
Comment (shawnbusuttil) [inline apps/browser-extension-wallet/src/views/browser-view/features/wallet-setup/components/HardwareWalletFlow.tsx:11]: Consider keeping the component names the same. But only change the filename so then it's easier after we sync with multi wallet

Comment (shawnbusuttil) [reply apps/browser-extension-wallet/src/views/browser-view/features/wallet-setup/components/HardwareWalletFlow.tsx:11]: Actually just changing the directory should be enough if we put everything in there.

File: packages/core/src/ui/components/RevampedWalletSetup/RevampedWalletSetupSelectAccountsStep.tsx [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+  const [selectedAccount, setSelectedAccount] = useState<string | undefined>('0');
+  const [walletName, setWalletName] = useState(INITIAL_WALLET_NAME);
+  const [isDirty, setIsDirty] = useState(false);
+  const { t } = useTranslate();
+
+  const isNameValid = useMemo(() => validateName(walletName), [walletName]);
+
+  const validationError = useMemo(
+    () => !isNameValid && t('core.walletSetupSelectAccountsStep.maxCharacters'),
+    [isNameValid, t]
```
Comment (shawnbusuttil) [inline packages/core/src/ui/components/RevampedWalletSetup/RevampedWalletSetupSelectAccountsStep.tsx:36]: Do we really need memo here? The error won't show anyway in the jsx if the name isn't valid.

**Review by shawnbusuttil — COMMENTED** [review]
Left a few comments.

**Review by danielmain — CHANGES_REQUESTED** [review]
Small change

File: apps/browser-extension-wallet/src/views/browser-view/features/wallet-setup/components/HardwareWalletFlow.tsx (hunk 1 above)
Comment (greatertomi) [reply apps/browser-extension-wallet/src/views/browser-view/features/wallet-setup/components/HardwareWalletFlow.tsx:11]: Leaving the component name unchanged would through an error because 2 different components with the same name are exported. I tried this before, but didn't work.

File: packages/core/src/ui/components/RevampedWalletSetup/RevampedWalletSetupSelectAccountsStep.tsx (hunk 2 above)
Comment (greatertomi) [reply packages/core/src/ui/components/RevampedWalletSetup/RevampedWalletSetupSelectAccountsStep.tsx:36]: Copied this directly from the previous codebase, but would check this.

Comment (shawnbusuttil) [reply packages/core/src/ui/components/RevampedWalletSetup/RevampedWalletSetupSelectAccountsStep.tsx:36]: But this in plain english is saying "only store this variable if name is not valid". :S

File: apps/browser-extension-wallet/src/views/browser-view/features/wallet-setup/components/HardwareWalletFlow.tsx (hunk 1 above)
Comment (greatertomi) [reply apps/browser-extension-wallet/src/views/browser-view/features/wallet-setup/components/HardwareWalletFlow.tsx]: Renamed the files on 74de25d

File: packages/core/src/ui/components/RevampedWalletSetup/RevampedWalletSetupSelectAccountsStep.tsx (hunk 2 above)
Comment (greatertomi) [reply packages/core/src/ui/components/RevampedWalletSetup/RevampedWalletSetupSelectAccountsStep.tsx]: Fixed on 6843ba4

**Review by danielmain — APPROVED** [review]
Great job @greatertomi

**Review by oldGreg5 — APPROVED** [review]
